### DISCUSSION TITLE
Consistent `github.workspace` directory

### DISCUSSION THREAD

**Discussion by Lnc** [ChristopherHX/github-act-runner#192] [Q&A]
Hi,

With the native GitHub runner, the `github.workspace` remains consistent across runs and looks like `/home/user/actions-runner/_work/project/project`.

With github-act-runner, it looks like `/home/user/.cache/act/360790f1e77d3d02/hostexecutor`, and the `360790f1e77d3d02` part changes with each run.

Is it possible to have the same directory for all runs?

Thank you.

**Comment by ChristopherHX** [comment]
Hi, 
What is your usecase for wanting a consistent workspace?

Currently there is no configuration option to change this, but this could be added easyly with a small change.

Do you want to preserve the workdir content of the previous run?
Do you want this scoped by repository name like in actions/runner?

**Comment by Lnc** [comment]
Hi,
With the current setup, ccache has 100% cache misses. (I just need the same path.)

**Comment by ChristopherHX** [comment]
I have a problem to decide the correct solution here..., my act fork has been updated to allow changing this.

Some options I had in mind
- Make this optional by implementing `--work` during configure and store a json field in config
- Make this apply to all new configurations by always choose the random workspace folder during configure
- Make act options available to configure via a config file

What do you think?

Architectureally I have split code interacting with act and the code with github to allow replacing the execution engine on the fly.

**Comment by Lnc** [comment]
Sorry for the late response, and thank you very much for your answer.
In the classic GitHub Actions runner registration, we can choose the working directory (if I remember correctly).
If you choose this option, I can think of three choices: random, random for each run, and custom (I will use the first or third one).
